### DISCUSSION TITLE
Solve mistakes in documents uploaded to workflowitems

### DISCUSSION THREAD

**Discussion by georgimld** [openkfw/TruBudget#1540] [Ideas]
Sometimes users make a mistake when uploading a document to a workflowitem, and upload the wrong document. It would perhaps be useful to somehow fix this problem. 

- One suggestion would be to "delete" this document - however since TruBudget is a blockchain-based application the concept of "deleteion" does not really exist, so even though documents are stored in a separate blob storage, the referencess, ids and hashes of these files are still stored on chain, so it's not technically possible to remove the reference of the documents.
- Another way to solve this would be to mark the document as a "mistake", add an event to the chain to mark the document and apply this logic when listing documents.

Please feel free to discuss the above mentioned idea or come up with additional solutions.

**Comment by mayrmartin** [comment]
I think adding a "delete" option defeats the purpose of using a blockchain-based system. In my opinion, a possible solution would be an additional step when uploading the document where the user can view the uploaded document before submitting/uploading it to TruBudget. The check could be done by adding a Pop-Up where the document that shall be uploaded must be checked, the user has to confirm i.e. with a checkbox, that the document is the correct one. Important here is that the user knows that the displayed document has not yet been uploaded to TruBudget.

**Comment by jzakotnik** [comment]
I would have thought an option would be to
- Mark the document as "deleted" (not "mistake", this could have consequences for the user in the organisation, e.g. disciplinary measures)
- Delete the document from the store, invalidating the link

This way it remains transparent that the document was there, it is clear that it was removed on purpose (regardless of being a mistake or any other/unknown reason).

**Comment by jzakotnik** [comment]
One feedback from a project: If a mistake is done on a workflow item, it is also possible to remove all access rights from that workflow item, then it's displayed as "redacted" to everyone

**Comment by LukasKM** [comment]
My preference would be to implement the suggestion above with marking the document as "deleted" and removing it from the store: 

-     Mark the document as "deleted" 
-     Delete the document from the store, invalidating the link. 

This should be possible until the WF item has been validated/rejected.
